pytorch_lightning/trainer/trainer.py

```diff
@@ -2292,6 +2291,17 @@ def terminate_on_nan(self, val: bool) -> None:
         )
         self._terminate_on_nan = val  # : 212
 
+    def convert_to_lightning_optimizers(self):
+        def _convert_to_lightning_optimizer(trainer, optimizer):
```

Comment:
Yes agree, something needs to change here because the same method cannot be in the "mixin" and on the Trainer, otherwise one of the two gets ignored.

Reply:
> Yes agree, something needs to change here because the same method cannot be in the "mixin" and on the Trainer, otherwise one of the two gets ignored.

Good point. Alternatively we can just mark this method as private. It might make sense for it to be in the trainer since it is modifying state on the trainer. But I don't feel too strongly either way

Reply:
> Yes agree, something needs to change here because the same method cannot be in the "mixin" and on the Trainer, otherwise one of the two gets ignored.

Good point. Alternatively we can just mark this method as private. It might make sense for it to be in the trainer since it is modifying state on the trainer. But I don't feel too strongly either way

Reply:
sounds good. and would you move the inner function to the optimizer.py as well as @tchaton suggested?